packages/web3-eth2-beaconchain/src/index.ts

```diff
@@ -0,0 +1,582 @@
+            });
+        } catch (error) {
+            throw Error(
+                `Error getting state root from state Id: ${error.message}`
+            );
+        }
+    }
+
+    /**
+     * Returns Fork object for state with given stateId.
+     * @param {StateId} stateId State identifier, can be "head", "genesis", "finalized", "justified", slot, hex string encoded stateRoot with 0x prefix
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @returns {Promise} A fork object
+     */
+
+    async getStateFork(
+        stateId: StateId,
+        requestArguments?: Partial<Eth2RequestArguments>
+    ): Promise<RpcStringResult> {
+        try {
+            return await this.provider.request({
+                ...requestArguments,
+                endpoint: `states/${stateId}/fork`,
+            });
+        } catch (error) {
+            throw Error(`Error getting state fork from Id: ${error.message}`);
+        }
+    }
+
+    /**
+     * Get finality checkpoints for state with given 'stateId'. In case finality is not yet achieved, checkpoint should return epoch 0 and ZERO_HASH as root
+     * @param {StateId} stateId State identifier, can be "head", "genesis", "finalized", "justified", slot, hex string encoded stateRoot with 0x prefix
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @returns {Promise} Returns finality checkpoints for state with given 'stateId'. If finality is not yet achieved, checkpoint should return epoch 0 and ZERO_HASH as root.
+     */
+
+    async getFinalityCheckpoints(
+        stateId: StateId,
+        requestArguments?: Partial<Eth2RequestArguments>
+    ): Promise<RpcStringResult> {
+        try {
+            return await this.provider.request({
+                ...requestArguments,
+                endpoint: `states/${stateId}/finality_checkpoints`,
+            });
+        } catch (error) {
+            throw Error(`Error getting finality checkpoints: ${error.message}`);
+        }
+    }
+
+    /**
+     * Get a list of validators from state
+     * @param {StateId} stateId State identifier, can be "head", "genesis", "finalized", "justified", slot or hex string encoded stateRoot with 0x prefix.
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @param {[string]} id (Optional) Array of ids, Either a hex string encoded public key (with 0x prefix) or validator index
+     * @param {Status} status (Optional) Validator status specification https://hackmd.io/ofFJ5gOmQpu1jjHilHbdQQ
+     * @returns {Promise} Array of the validators specified with balance, status and index
+     */
+
+    async getValidators(
+        stateId: StateId,
+        id?: [string],
```

Comment:
```suggestion
        id?: string[],
```

In typescript you denote an array of strings as `string[]`

`[string]` means a fixed size array with the first index being of type string